teamscale-maven-plugin/src/main/java/com/teamscale/tia/maven/TiaMojoBase.java

```diff
@@ -0,0 +1,290 @@
+package com.teamscale.tia.maven;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.conqat.lib.commons.filesystem.FileSystemUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Base class for TIA Mojos. Provides all necessary functionality but can be subclassed to change the partition.
+ * <p>
+ * For this plugin to work, you must either
+ *
+ * <ul>
+ *     <li>Make Surefire and Failsafe use our JUnit 5 test engine</li>
+ *     <li>Send test start and end events to the Java agent themselves</li>
+ * </ul>
+ * <p>
+ * To use our JUnit 5 test engine, you must declare it as a dependency of the maven-surefire-plugin and/or the
+ * maven-failsafe-plugin. Example:
+ *
+ * <pre>{@code
+ * <plugin>
+ *     <groupId>org.apache.maven.plugins</groupId>
+ *     <artifactId>maven-surefire-plugin</artifactId>
+ *     <version>3.0.0-M5</version>
+ *     <dependencies>
+ *         <dependency>
+ *             <groupId>com.teamscale</groupId>
+ *             <artifactId>teamscale-surefire-provider</artifactId>
+ *             <version>23.2.0</version>
+ *         </dependency>
+ *     </dependencies>
+ * </plugin>
+ * }</pre>
+ * <p>
+ * To send test events yourself, you can use our TIA client library (Maven coordinates: com.teamscale:tia-client).
+ * <p>
+ * The log file of the agent is written to {@code ${project.build.directory}/tia/agent.log}.
+ */
+public abstract class TiaMojoBase extends AbstractMojo {
```

Comment:
What is the `defaultPhase of this goal? Would expect `initialize`. Please make this explicit with the `@Mojo` annotation.

Reply:
This is not an actual goal - thus the class is abstract. It just encapsulates the common logic shared between the two actual Mojos (which both have the defaultPhase specified). Thus also the `...Base` naming. I don't think it makes sense to give this class a defaultPhase, especially since the two subclasses have different defaultPhase annotations.